src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts

```diff
@@ -320,7 +320,9 @@ export class ExtensionsListView extends CollapsibleView {
 				}
 
 				return this.extensionsWorkbenchService.queryGallery(assign(options, { names, pageSize: names.length }))
-					.then(pager => new PagedModel(pager));
+					.then(pager => this.addRecommendationInfo(new PagedModel(pager || []), {
+						source: 'fileType'
+					}));
```

Comment:
@sandy081 On second thoughts, using queryOptions is not right. The recommendationInfo has nothing to do with the actual query. It is a property of the extension that is determined by VS Code and not the galleryService.

How about having some kind of setter on the `telemetryData` property?